event-algo.bs

```diff
@@ -2073,10 +2183,16 @@ Issue: Should these be moved into the Clipboard spec?
     1. Let |data| = the text on the clipboard being pasted
     1. Let |result| = <a>fire an InputEvent</a> with "beforeinput", "insertFromPaste" and |data|
 
-    1. If |result| is true, then
+    1. If |result| is false, then return.
+
+    1. Set |result| = <a>fire a TextEvent</a> with "textInput", and |data|
+
+        Note: The "textInput" event is obsolete.
+
+    1. If |result| is false, then return.
 
-       1. Paste clipboard contents into DOM target element
-       1. <a>Fire an InputEvent</a> with "input", "insertFromPaste" and |data|
+    1. Paste clipboard contents into DOM target element
+    1. <a>Fire an InputEvent</a> with "input", "insertFromPaste" and |data|
 
 </div><!-- algorithm -->
```

Comment:
If `inputType` of `beforeinput` is `insertLineBreak`, `textInput` is fired with `data="\n"` on Chrome.

Reply:
If `inputType` of `beforeinput` is `insertCompositionText` **and** the composition is being committed, `textInput` is fired with `data=beforeinput.data`. On Chrome, `isComposing` of `beforeinput` and `input` are `true`. Unfortunately, I think that there is no way to define this behavior with attributes of the event. See also <https://https://github.com/w3c/uievents/issues/202>.

Reply:
Odd, in the case of DnD, Chrome dispatches `textInput` immediately after `drop`, but before `beforeinput` whose `inputType` is (`deleteByDrag` and) `insertFromDrop`. Additionally, the event target of `textInput` is not the editing host.

I think that Gecko cannot emulate this behavior.

Reply:
`textInput` when composition is committed: 2a3cbd8

Reply:
`insertLineBreak` and `insertParagraph` cases fixed in f36ff37

Reply:
(I'm still waiting the DnD definition and adding `insertFromPasteAsQuotation` after `insertFromPaste` (I mean `s/"insertFromPaste"/"insertFromPaste" or "insertFromPasteAsQuotation"/`)

Reply:
Filed https://github.com/w3c/uievents/issues/367 for DnD.